### PR TITLE
lestarch: initial style guidelines

### PR DESCRIPTION
**Pull request by LeStarch**
| | |
|:---|:---|
|**_Originating Project/Creator_**| |
|**_Affected Component_**|  |
|**_Affected Architectures(s)_**|  |
|**_Related Issue(s)_**|  |
|**_Has Unit Tests (y/n)_**|  |
|**_Builds Without Errors (y/n)_**|  |
|**_Unit Tests Pass (y/n)_**|  |
|**_Documentation Included (y/n)_**|  |

---
## Change Description

Adds in style guide.

## Rationale

Opening up other contributors.

## Testing/Review Recommendations

Just documentation.

## Future Work

More documentation.

### CODE REVIEW

File: docs/UsersGuide/dev/code-style.md [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+Many of F´ constructs are required to inherit from a autocoded class that then inherits from framework base classes.
+This can present a problem as this precludes inheritance from other structures. Thus limited multiple-inheritance is
+permitted. Basic guidelines are presented here, however; use of multiple-inheritance will likely be discussed during
+the submission review.
+
+1. Only use multiple-inheritance when absolutely required
+2. Avoid inheritance from more than one hand-coded class
+3. Avoid inheritance from hand-coded descendants of autocoded classes and parents (e.g. `Fw::PortBase` and
+   `Fw::PassiveComponentBase`) not. i.e avoid inheritance from other Components, Ports, or classes that do.
+
```
Comment (timcanham) [inline docs/UsersGuide/dev/code-style.md:112]: 4. Avoid multiple inheritance paths that derive from a common base class (the diamond problem): https://www.cprogramming.com/tutorial/virtual_inheritance.html

**Review by timcanham — CHANGES_REQUESTED** [review]

**Review by timcanham — APPROVED** [review]

**Comment by LeStarch** [comment]
Noting @timcanham approval.  Pushing to fix spelling and nothing more.

File: docs/UsersGuide/dev/code-style.md [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+10. Do not use `Os::Task::delay` to synchronize between threads
+11. Explicit enumeration values should be specified for all values or none at all
+
+### C++11 Usage
+
+F´ has been updated to support C++11. Some C++11 implementations are incomplete requiring some deviations from C++11.
+Thus, there are some things that developers should be aware of when using C++11 features:
+
+1. Many C++11 features use templates under-the-hood.  Prefer `Fw` and `Os` implementations for these features.
+2. Use of `std::atomic` is known to not work is some cases. Use is avoided.
```
Comment (Joshua-Anderson) [inline docs/UsersGuide/dev/code-style.md:66]: Spelling fix: s/is/in/

**Comment by LeStarch** [comment]
Fixed issues, changes since review approval:

1. spelling fix for new webaddress
2. is->in as requested.

**Pull request merged**